core/scripts/vrfv2/testnet/util.go

```diff
@@ -159,29 +158,49 @@ func decreasingBlockRange(start, end *big.Int) (ret []*big.Int, err error) {
 	return
 }
 
-func getRlpHeaders(ec *ethclient.Client, blockNumbers []*big.Int) (headers [][]byte, err error) {
+func getRlpHeaders(env helpers.Environment, blockNumbers []*big.Int) (headers [][]byte, err error) {
 	headers = [][]byte{}
 	for _, blockNum := range blockNumbers {
-		// Get child block since it's the one that has the parent hash in it's header.
-		h, err := ec.HeaderByNumber(
-			context.Background(),
-			new(big.Int).Set(blockNum).Add(blockNum, big.NewInt(1)),
-		)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get header: %+v", err)
-		}
-		rlpHeader, err := rlp.EncodeToBytes(h)
-		if err != nil {
-			return nil, fmt.Errorf("failed to encode rlp: %+v", err)
+		// Avalanche block headers are special, handle them by using the avalanche rpc client
+		// rather than the regular go-ethereum ethclient.
+		if helpers.IsAvaxNetwork(env.ChainID) {
```

Comment:
I tried to see about a cleaner way to do this (making an abstractClient interface) but i guess the Headers being different types is tough to get around. LGTM.

Reply:
Approved pending tests

Reply:
Yeah, I tried to do that as well, but pretty much all the types are under a different namespace (e.g `*types.Transaction`, not just `*types.Header`) so this seemed like the only way to get around it for now.